kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -154,6 +156,10 @@ public void writeRows(PartitionedTableId table,
           // rate limit exceeded error
           logger.warn("Rate limit exceeded for table {}, attempting retry", table);
           retryCount++;
+        } else if (err.getCode() == UNKNOWN_CODE && err.getCause() != null && err.getCause() instanceof IOException){
+          //Retry when IO exceptions occur
+          logger.warn("IO Exception: {}, attempting retry", err.getCause().getMessage());
+          retryCount++;
```

Comment:
The logic here seems mostly fine, but I'm wondering if we can move it into the new `BigQueryErrorResponses` class introduced in https://github.com/confluentinc/kafka-connect-bigquery/pull/145? I think a new `isUnspecifiedIOError` method could work?

It'd also be nice to see a basic unit test for this logic if it's not too much trouble.